tests/integration/test_diag_indices.py

```diff
@@ -20,23 +20,106 @@
 import cunumeric as cn
 
 
+def test_diag_indices_default_ndim():
+    np_res = np.diag_indices(10)
+    cn_res = cn.diag_indices(10)
+    assert np.array_equal(np_res, cn_res)
+
+
 @pytest.mark.parametrize("ndim", range(0, LEGATE_MAX_DIM + 1))
-def test_diag_indices(ndim):
+def test_diag_indices_basic(ndim):
     np_res = np.diag_indices(10, ndim)
     cn_res = cn.diag_indices(10, ndim)
     assert np.array_equal(np_res, cn_res)
 
 
+@pytest.mark.parametrize("n", [0, 0.0, 1, 10.5])
+@pytest.mark.parametrize("ndim", [-4, 0, 1])
+def test_diag_indices(n, ndim):
+    np_res = np.diag_indices(n, ndim)
+    cn_res = cn.diag_indices(n, ndim)
+    assert np.array_equal(np_res, cn_res)
+
+
+class TestDiagIndicesErrors:
+    @pytest.mark.parametrize("n", [-10.5, -1])
+    def test_negative_n(self, n):
+        msg = "Invalid shape"
+        print("[robinw]: test-1")
+        with pytest.raises(ValueError, match=msg):
+            cn.diag_indices(n)
+
+    @pytest.mark.xfail
+    @pytest.mark.parametrize("n", [-10.5, -1])
+    def test_negative_n_compare_with_numpy(self, n):
```

Comment:
FYI I think we should settle on a common naming convention for all the `xfail` diverge-from-numpy tests. A `_compare_with_numpy` suffix could be OK, but maybe it would not be bad to make them even a little more prominent. What do you think about `test_negative_n_DIVERGENCE`?

Reply:
Good suggestion. Updated.